csharp/src/Ice/Communicator.cs

```diff
@@ -840,7 +834,7 @@ public async ValueTask<IObjectPrx> CreateAdminAsync(ObjectAdapter? adminAdapter,
                     throw new CommunicatorDisposedException();
                 }
 
-                if (_adminAdapter != null)
+                if (_adminAdapter != null || _getAdminTask != null)
```

Comment:
Is this check correct? we don't hold the mutex while calling `GetAdminAsync` so there is a chance that a second call to `CreateAdminAsync` enters here before  `_getAdminTask` was created

Reply:
I don't understand your concern. _getAdminTask is always read and updated with _mutex locked.

One potential race condition here would be 2 concurrent calls to CreateAdminAsync that end up calling GetAdminAsync, and both succeed when in theory one of them should fail. For me, that's not a concern.

Reply:
The concern is that the call that success use the identity set by the one that fail
```
Thread 1 call CreateAdminAsync and is awaiting GetAdminAsync (mutex not locked)
Thread 2 calls CreateAdminAsync and updates the _adminIdentity

Thread 1 GetAdminAsync returns and the you are using the identity set by thread 2
```

Reply:
Fixed by reworking the code.